src/sentry/tagstore/snuba/backend.py

```diff
@@ -157,21 +162,50 @@ def __get_tag_keys_for_projects(
         limit=1000,
         keys=None,
         include_values_seen=True,
+        use_cache=True,
         **kwargs
     ):
-        filters = {"project_id": projects}
+        filters = {"project_id": sorted(projects)}
         if environments:
-            filters["environment"] = environments
+            filters["environment"] = sorted(environments)
         if group_id is not None:
             filters["group_id"] = [group_id]
         if keys is not None:
-            filters["tags_key"] = keys
+            filters["tags_key"] = sorted(keys)
         aggregations = [["count()", "", "count"]]
 
         if include_values_seen:
             aggregations.append(["uniq", "tags_value", "values_seen"])
         conditions = []
 
+        should_cache = (
+            random.random() <= options.get("snuba.tagstore.cache-tagkeys-rate")
```

Comment:
This won't fly. Let's impagine we start with 10%, if the rollout percentage is not consistent with the cache key we will skew the hit rate towards 0 (if I have two subsequent requests for the same cache key and the rollout check passes in one case and not in the other you would be missing a cache hit). 
To avoid this you can compute the cache key first and then hash that key into one value that you would use instead of the random generator so that given a key, it will consistently fail the check or consistently pass it.